### PR TITLE
DevOps

### PR DESCRIPTION
**Pull request by whipermr5**

### CODE REVIEW

File: contents/devops/DevOps.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+This includes using version control tools, following a process when developing, frequently merging code with a central branch, and going through a code review process.
+
+#### Relevant Tools
+
+- Git
+- GitHub
+
+### Build & Test Automation
+
+Involves using continuous integration tools that automatically build and test your software upon every push, so you don't have to do it yourself.
```
Comment (li-kai) [inline contents/devops/DevOps.md]: Perhaps can explain `push` as git push. However, it's not just that too, is it? Would be nice to mention that the tools can run for pull requests, etc.

File: contents/devops/DevOps.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+#### Relevant Tools
+
+- Gradle
+- Travis CI
+- Jenkins
+
+### Deployment & Infrastructure
+
+This is about getting your software to run in a live server environment so that it is publicly accessible to end users. Also involves packaging your app such that it can be deployed in different environments.
```
Comment (li-kai) [inline contents/devops/DevOps.md]: Nitpick about the grammar (only one I can find, nice job!)

> Also involves packaging

Should be

> It also involves packaging

**Review by li-kai — COMMENTED** [review]
I really like the content. It's well structured.

I wish you could go more into the relevant tools and maybe explain if they have any differences in their utility, to prevent beginners from wasting their time exploring every tool.
E.g. Travis CI and Jenkins are pretty similar, Gradle is the odd one out. 
E.g. DigitalOcean being the odd one out among all the deployment tools.

**Comment by damithc** [comment]
@whipermr5 let me know after approved by peer reviewer

File: contents/devops/DevOps.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+These can be roughly summarised into four major aspects:
+
+### Code Development & Review
+
+This includes using version control tools, following a process when developing, frequently merging code with a central branch, and going through a code review process.
+
+#### Relevant Tools
+
+- Git
+- GitHub
```
Comment (damithc) [inline contents/devops/DevOps.md]: May be you can mention these as examples. There are other vialble alternatives to these.

File: contents/devops/DevOps.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+This is about getting your software to run in a live server environment so that it is publicly accessible to end users. Also involves packaging your app such that it can be deployed in different environments.
+
+#### Relevant Tools
+
+- DigitalOcean
+- Vagrant
+- Docker
+- Ansible
+- Puppet
```
Comment (damithc) [inline contents/devops/DevOps.md]: Try to add:
1. Link to the website
2. A brief overview of the tool

**Review by damithc — COMMENTED** [review]
Good work @whipermr5 
Some comments added. Revise as you see fit.
Adding more content and/or getting more peer reviews is welcome but not required. 
I will be merging end of week 12

File: contents/devops/DevOps.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+## Useful Links
+
+### Quick References
+- [What is DevOps?](https://aws.amazon.com/devops/what-is-devops/) - a concise article by Amazon Web Services
+- [The Essential DevOps Terms You Need to Know](https://blog.xebialabs.com/2016/03/21/essential-devops-terms/) - understand the most commonly used terms in the industry
+
+### Further Reading
+- [Best Guide to Getting Started In DevOps](https://blog.xebialabs.com/2015/04/23/best-guide-to-getting-started-in-devops/) - recommends various places you can read up more on the topic
+- [9½ Simple Steps On How To Start With DevOps Today](https://devops.com/9%C2%BD-simple-steps-start-devops-today/) - includes practical suggestions like using a code static analysis tool
+- [A Pragmatic Guide to Getting Started with DevOps](http://www.ca.com/us/lpg/~/media/Files/eBooks/a-pragmatic-guide-to-getting-started-with-devops.pdf) - more of a management point of view but interesting
```
Comment (mauris) [inline contents/devops/DevOps.md]: a good book about devops: https://books.google.com.sg/books/about/The_Devops_Handbook.html?id=XrQcrgEACAAJ&redir_esc=y

**Pull request merged**